trace-dispatcher/src/Cardano/Logging/Formatter.hs

```diff
@@ -0,0 +1,262 @@
+  pure $ Trace (T.arrow trr)
+ where
+    mkTracer = T.emit $
+      \ case
+        (lc, Nothing, v) ->
+          let metrics = asMetrics v
+          in T.traceWith tr (lc { lcNamespace = application : lcNamespace lc}
+                                , Nothing
+                                , FormattedMetrics metrics)
+        (lc, Just ctrl, v) ->
+          let metrics = asMetrics v
+          in T.traceWith tr (lc { lcNamespace = application : lcNamespace lc}
+                                , Just ctrl
+                                , FormattedMetrics metrics)
+
+-- | Format this trace as TraceObject for the trace forwarder
+forwardFormatter
+  :: forall a m . (LogFormatting a, MonadIO m)
+  => Text
+  -> Trace m FormattedMessage
+  -> m (Trace m a)
+forwardFormatter application (Trace tr) = do
+  hn <- liftIO getHostName
+  let trr = mkTracer hn
+  pure $ Trace (T.arrow trr)
+ where
+    mkTracer hn = T.emit $
+      \ case
+        (lc, Nothing, v) -> do
+          thid <- liftIO myThreadId
+          time <- liftIO getCurrentTime
+          let fh = forHuman v
+              details = fromMaybe DNormal (lcDetails lc)
+              fm = forMachine details v
+              nlc = lc { lcNamespace = application : lcNamespace lc}
+              to = TraceObject {
+                      toHuman     = if fh == "" then Nothing else Just fh
+                    , toMachine   = if fm == mempty then Nothing else
+                                    Just $ decodeUtf8 (BS.toStrict (AE.encode fm))
+                    , toNamespace = lcNamespace nlc
+                    , toSeverity  = fromMaybe Info (lcSeverity lc)
+                    , toDetails   = fromMaybe DNormal (lcDetails lc)
+                    , toTimestamp = time
+                    , toHostname  = hn
+                    , toThreadId  = (pack . show) thid
+                  }
+          T.traceWith tr ( nlc
+                         , Nothing
+                         , FormattedForwarder to)
+        (lc, Just ctrl, _v) -> do
+          thid <- liftIO myThreadId
+          time <- liftIO getCurrentTime
+          let nlc = lc { lcNamespace = application : lcNamespace lc}
+              to  = TraceObject {
+                      toHuman     = Nothing
+                    , toMachine   = Nothing
+                    , toNamespace = lcNamespace nlc
+                    , toSeverity  = fromMaybe Info (lcSeverity lc)
+                    , toDetails   = fromMaybe DNormal (lcDetails lc)
+                    , toTimestamp = time
+                    , toHostname  = hn
+                    , toThreadId  = (pack . show) thid
+                  }
+          T.traceWith tr ( nlc
+                         , Just ctrl
+                         , FormattedForwarder to)
+
+-- | Format this trace for human readability
+-- The boolean value tells, if this representation is for the console and should be colored
+-- The text argument gives the application name which is prepended to the namespace
```

Comment:
I think it's better style in general to put the comments on the actual arguments rather than in the function description: https://haskell-haddock.readthedocs.io/en/latest/markup.html#function-arguments

Reply:
I've not used this style so far, but I will consider it for the future.